### PR TITLE
EKS module fixes terraform

### PR DESCRIPTION
**Pull request by DaMandal0rian**
## **User description**
This PR adds a ssh key secret to secrets manager for declarative deployments and accessing github repo with argocd.
It also makes some fixes to the EKS root module and enabled addons.

___

## **Type**
enhancement, documentation

___

## **Description**
- Added Terraform cloud configuration for EKS blue and green environments.
- Introduced TLS provider in various configurations.
- Updated default variables across EKS modules and templates.
- Implemented SSH key pair management in AWS Secrets Manager.
- Refined EKS cluster addons and bootstrap configurations.
- Added ApplicationSets for bootstrap addons and workloads in ArgoCD.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><details><summary>5 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>backend.tf</strong><dd><code>Add Terraform Cloud Configuration for EKS Blue</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/eks-blue/backend.tf
<li>Added Terraform cloud configuration for workspace <br><code>subspace-aws-managed-k8s</code> in organization <code>subspace-sre</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-9ea05d843bc6f0048a3418ded9800de38f577b0bef838d8c38174dd622160580">+9/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>providers.tf</strong><dd><code>Include TLS Provider in EKS Blue</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/eks-blue/providers.tf
- Added `tls` provider with version `4.0.5`.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-47396f3824110bbefafb1c18ef0f02bf39ec05f29d202b9a9861a1dd66eafd9a">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>providers.tf</strong><dd><code>Include TLS Provider in EKS Green</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/eks-green/providers.tf
- Added `tls` provider with version `4.0.5`.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-229323b531df793c751b5f5c2c541661d9700b07e64abbeef3dadb3f30133e9f">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>providers.tf</strong><dd><code>Add TLS Provider to Network Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/network/providers.tf
- Introduced `tls` provider.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-580e81029ee8303a0da423aa9d0866cf1c85a1876988878f0e01f7df254e2f58">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>versions.tf</strong><dd><code>Include TLS Provider in Network Versions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/network/versions.tf
- Added `tls` provider with version `4.0.5`.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-69020205d03f69fb4662071ed914d0b0348de7759ac23138bd15248ff68a3967">+4/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Enhancement</strong></td><td><details><summary>9 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>variables.tf</strong><dd><code>Update Default Variables for EKS Blue</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/eks-blue/variables.tf
<li>Changed default value for <code>environment_name</code> to <code>cluster</code>.<br> <li> Updated <code>aws_secret_manager_git_private_ssh_key_name</code> default to <br><code>github-subspace-ssh-key-private</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-41f96efaa7109d6caf280d0fda851793744c9663a156cab9ec20b76804656b56">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>variables.tf</strong><dd><code>Update Default Variables for EKS Green</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/eks-green/variables.tf
<li>Changed default value for <code>environment_name</code> to <code>cluster</code>.<br> <li> Updated <code>aws_secret_manager_git_private_ssh_key_name</code> default to <br><code>github-subspace-ssh-key-private</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-82852f1800ce2fa6630a6fd5852dd5ab83cb6f8986a98ce2792ad20d2db78ea3">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.tf</strong><dd><code>Enhance ACM Certificate Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/network/main.tf
<li>Added alternative names for ACM certificate.<br> <li> Set <code>wait_for_validation</code> to <code>false</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-9a53583c0fb470e0caf937b2459bf9506a6737031a32d9ae07e3d9c22ea615a6">+3/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>secrets.tf</strong><dd><code>Implement SSH Key Management in Secrets Manager</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/network/secrets.tf
<li>Created resources for SSH key pair and stored in AWS Secrets Manager.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-c361f06b5618ab397b4efab481ebb39ad8882121288d69ec36e52b9e39f7002f">+28/-0</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>variables.tf</strong><dd><code>Update Network Configuration Default Variables</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

eks/network/variables.tf
- Updated default `environment_name` to `subspace`.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-42adf3def8a1c9d36b02270cae28e1f18b024008d5a22bd51811cd4dfe5851c8">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.tf</strong><dd><code>Refine EKS Cluster Addons and Bootstrap Configurations</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/terraform/aws/eks/eks_cluster/main.tf
<li>Adjusted AWS and OSS addons configurations.<br> <li> Commented out and added new addons.<br> <li> Updated paths for bootstrap files.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-fcef8deb7deeefc9583c809c28021b808266930c93dff5e1a83cf11767c81ddc">+66/-40</a>&nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>variables.tf</strong><dd><code>Update EKS Cluster Template Variables</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/terraform/aws/eks/eks_cluster/variables.tf
<li>Changed default <code>environment_name</code> to <code>subspace</code>.<br> <li> Updated <code>aws_secret_manager_git_private_ssh_key_name</code> default to <br><code>github-subspace-ssh-key</code>.<br>

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-229cc1674039179c59b0a2fa6b290e7b549ee06136504b05b7db624c7821910b">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>addons.yaml</strong><dd><code>Add Bootstrap Addons ApplicationSet for ArgoCD</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/terraform/aws/eks/bootstrap/addons.yaml
- Added ApplicationSet for bootstrap addons with ArgoCD.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-ba9af612b08237738f662cf04a503406d8ca74d3b69d7106461eb89b010964aa">+33/-1</a>&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>workloads.yaml</strong><dd><code>Add Bootstrap Workloads ApplicationSet for ArgoCD</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/terraform/aws/eks/bootstrap/workloads.yaml
- Added ApplicationSet for bootstrap workloads with ArgoCD.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-c5aa82df4ac430d0fff75f0ae63c6bc45a2af343953c53cfdf4919dd1fb5067b">+68/-1</a>&nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr><tr><td><strong>Formatting</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>versions.tf</strong><dd><code>Formatting Adjustment in EKS Cluster Versions</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

templates/terraform/aws/eks/eks_cluster/versions.tf
- Minor formatting adjustment.

</details>
    

  </td>
  <td><a href="https://github.com/subspace/infra/pull/285/files#diff-b054d9c5f36d33ad0f83a8cbf7f9dafd54df99007069d8d46942f5b45b0be8ca">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: templates/terraform/aws/eks/eks_cluster/main.tf [hunk 1]
```diff
@@ -42,40 +42,49 @@ locals {
...
+    #enable_aws_privateca_issuer                  = true
+    #enable_cluster_autoscaler                    = true
+    enable_external_dns                 = true
+    enable_external_secrets             = true
+    enable_aws_load_balancer_controller = true
+    #enable_fargate_fluentbit                     = true
+    enable_aws_for_fluentbit = true
+    #enable_aws_node_termination_handler          = true
+    enable_karpenter = true
+    #enable_velero                                = true
```
Comment (vedhavyas) [inline templates/terraform/aws/eks/eks_cluster/main.tf]: why keep them if we dont need them ?

File: templates/terraform/aws/eks/eks_cluster/main.tf [hunk 2]
```diff
@@ -345,13 +366,15 @@ resource "kubernetes_secret" "git_secrets" {
...
+      url  = local.gitops_addons_url
+      # comment if you want to uses public repo wigh syntax "https://github.com/xxx" syntax, uncomment when using syntax "git@github.com:xxx"
       sshPrivateKey = data.aws_secretsmanager_secret_version.workload_repo_secret.secret_string
     }
     git-workloads = {
-      type          = "git"
-      url           = local.gitops_workloads_url
+      type = "git"
+      url  = local.gitops_workloads_url
+      # comment if you want to uses public repo wigh syntax "https://github.com/xxx" syntax, uncomment when using syntax "git@github.com:xxx"
```
Comment (vedhavyas) [inline templates/terraform/aws/eks/eks_cluster/main.tf]: I think we should just always use git@github format for both public and private repos. Not sure if there is any beneficial with or without ssh key

**Review by vedhavyas — COMMENTED** [review]
make sense. Left some questions

File: templates/terraform/aws/eks/eks_cluster/main.tf (hunk 1 above)
Comment (DaMandal0rian) [reply templates/terraform/aws/eks/eks_cluster/main.tf]: I've removed the ones that we will never use, but commented ones that we will or likely use later in the cluster.

**Review by vedhavyas — APPROVED** [review]

**Pull request merged**